### ISSUE TITLE
ogr2ogr --help returns 1

### ISSUE THREAD

**Issue by byezy** [OSGeo/gdal#10612]
### What is the bug?

ogr2ogr --help returns 1:

`$ ogr2ogr --help; echo $?
Usage: ogr2ogr [--help] [--help-general]
               [-skipfailures] [-append | -upsert] [-update]
...
...
               [-nomd] [-mo <META-TAG>=<VALUE>]... [-noNativeData]

Note: ogr2ogr --long-usage for full help.
1
`
This behavior seems inconsistent within this program other gdal utilities.

ogr2ogr --help-general and ogr2ogr --long-usage both return 0.

gdalinfo --help returns 0

Minor, but annoying when developing an API that reacts to non-zero exits. 

Or is there a reason behind this?

### Steps to reproduce the issue

`$ ogr2ogr --help; echo $?`

### Versions and provenance

```
$ ogr2ogr --version; echo $?
GDAL 3.8.4, released 2024/02/08
0
```

**Comment by agiudiceandrea** [comment]
It seems to me the issue doesn't occur using ogr2ogr 3.9.1 on Windows (OSGeo4W).

**Comment by rouault** [comment]
> Probably fixed by [64dfb95](https://github.com/OSGeo/gdal/commit/64dfb95b8d20aaa312677cfa10fc7ce51c648a2d).

indeed
